libfoundation/src/foundation-unicode.cpp

```diff
@@ -140,9 +142,29 @@ const UProperty MCUnicodePropToICUProp[] =
 
 ////////////////////////////////////////////////////////////////////////////////
 
+// This is unused, but needs to be declared to resolve link errors in libicu
+char icudt58_dat[] = "";
```

Comment:
I think we can replace this by just linking with sicudt.lib (Win) and variants on the other platforms. This would mean that a change in ICU version wouldn't mean we had to update the foundation source.